app/controllers/fhir/r4/api_controller.rb

```diff
@@ -310,14 +325,32 @@ def options
 
   private
 
-  # Current user account as authenticated via doorkeeper
+  # Current user account as authenticated via doorkeeper for user flow
   def current_resource_owner
-    User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
+    User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token&.resource_owner_id
+  end
+
+  # Client application that is currently using the API
+  def current_client_application
+    Doorkeeper::Application.find_by(id: doorkeeper_token.application_id) if doorkeeper_token.application_id.present?
   end
 
-  # Current user account can use api
-  def check_api_access
-    current_resource_owner&.can_use_api?
+  # Determine the patient data that is accessable by either the current resource owner
+  # (user flow) or the current client application (system flow).
+  def accessable_patients
+    # If there is a current resource owner (end user) that has api access enabled
+    if current_resource_owner.present? && current_resource_owner&.can_use_api?
+      # This will access all patients that the role has access to, if any
+      current_resource_owner.patients
+    # Otherwise if there NO resource owner and there is a found application, check for a valid associated jurisdiction id.
+    # The current resource owner check is to prevent unauthorized users from using it if the application happens to be registered for both workflows.
+    elsif !current_resource_owner.present? && current_client_application.present?
+      jurisdiction_id = current_client_application[:jurisdiction_id]
+      return if jurisdiction_id.nil? || !Jurisdiction.find_by(id: jurisdiction_id).present?
```

Comment:
This can be: `Jurisdiction.exists?(jurisdiction_id)`

Reply:
Done.